File: SPSP/SPSP.Models/Customer.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace SPSP.Models
{
    public partial class Customer
    {

        public int Id { get; set; }
        public int? UserAccountId { get; set; }
        public string Address { get; set; }
        public string Phone { get; set; }
        public int? PenaltyPoints { get; set; }
        public bool? Valid { get; set; }
        public virtual UserAccount UserAccount { get; set; }

    }
}

File: SPSP/UI/spsp_mobile/lib/models/customer.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:spsp_mobile/models/user_account.dart';

part 'customer.g.dart';

@JsonSerializable()
class Customer {
  int id;
  int? userAccountId;
  UserAccount? userAccount;
  String? address;
  String? phone;
  int? penaltyPoints;
  bool? valid;

  Customer(this.id, this.userAccountId, this.userAccount, this.address, this.phone,
      this.penaltyPoints, this.valid);

  factory Customer.fromJson(Map<String, dynamic> json) => _$CustomerFromJson(json);

  Map<String, dynamic> toJson() => _$CustomerToJson(this);
}
